### PR TITLE
switch to meson

### PR DESCRIPTION
**Pull request by Mic92**
This has several advantages

- explicit pkg-config dependencies and error messages if they are missing
- cross-platform installation process
- no hard-coded compiler -> also accepts cross-compilers/clang
- more hardening flags
- compile_command.json for language-servers/clang-tooling
- faster incremental rebuilds
- out-of-tree builds

### CODE REVIEW

File: test/test.c [hunk 1]
```diff
@@ -73,7 +75,7 @@ int test_files_match(char *example_name, char *expected_file_path,
 }
 
 int test_render(char *example) {
-  char *example_dir = "test/example/";
+  char *example_dir = TEST_EXAMPLE_DIR;
```
Comment (Mic92) [inline test/test.c:78]: This makes tests no longer crash if they are not started from the test directory.

**Comment by ryantm** [comment]
Are meson and ninja already used extensively in the minimal NixOS closure?

**Comment by Mic92** [comment]
> Are meson and ninja already used extensively in the minimal NixOS closure?

Yes. i.e. systemd, qemu, gtk and wayland
It is also quite portable. More portable than the average makefile for example.

**Comment by ryantm** [comment]
@Mic92 What's up with the triple underscores before the embedded assets?

**Comment by Mic92** [comment]
> @Mic92 What's up with the triple underscores before the embedded assets?

This is because xxd computes the name based on executable path. With meson there are out of tree builds enabled so the source file path becomes `../src/asset/$name` (xxd is called from `build`). I have not found a nice solution that would have not resolved in writing some ugly shell wrapper around xxd, so I decided that just adding changing the asset paths would be less friction/complex.

**Comment by Mic92** [comment]
One workaround would be to copy the source files to $PWD before executing `xxd` if you prefer that.

**Pull request merged**